clientconn_test.go

```diff
@@ -659,6 +659,8 @@ func (s) TestClientUpdatesParamsAfterGoAway(t *testing.T) {
 	}
 	defer lis.Close()
 	connected := make(chan struct{})
```

Comment:
This would be a little neater (taking advantage of a grpc internal feature):

```go
connected := grpcsync.NewEvent()
defer connected.Fire()
...
<-connected.Done()
...
connected.Fire()
```

Reply:
nice fix! updated @dfawley